src/firefly/js/ui/JoyToMUIThemeWrapper.jsx

```diff
@@ -0,0 +1,57 @@
+import React, {useEffect} from 'react';
+import {THEME_ID as MATERIAL_THEME_ID,
+    experimental_extendTheme as materialExtendTheme,
+    Experimental_CssVarsProvider as MaterialCssVarsProvider,
+    useColorScheme as useMaterialColorScheme
+} from '@mui/material/styles';
+import {CssVarsProvider as JoyCssVarsProvider} from '@mui/joy/styles';
+import {useColorMode, getTheme} from 'firefly/ui/FireflyRoot';
+
+
+/**
+ * This is needed to use MUI X components in JoyUI, otherwise they will throw errors because they expect MUI theme.
+ * Adapted from the example code present at https://mui.com/x/react-date-pickers/custom-field/#using-another-input
+ * and https://mui.com/joy-ui/integrations/material-ui/
+ *
+ * @param p
+ * @param p.children
+ */
+export function JoyToMUIThemeWrapper({children}) {
+    const {activeMode: joyMode} = useColorMode();
+    const joyTheme = getTheme();
+
+    // To sync MUI's mode with JoyUI's mode, when JoyUI's mode changes
+    const SyncThemeMode = ({ mode }) => {
+        const { setMode: setMaterialMode } = useMaterialColorScheme();
+        useEffect(() => {
+            setMaterialMode(mode);
+        }, [mode, setMaterialMode]);
+        return null;
+    };
+
+    // Override the default MUI appearance of wrapped component (esp font, color, etc.) with JoyUI theme
+    // Note: MUI and JoyUI theme objects are not identical by design, so following doesn't cover all cases
+    const materialTheme = materialExtendTheme({
+        typography: {
+            fontFamily: joyTheme.fontFamily.body
+        },
+        colorSchemes: Object.fromEntries(['light', 'dark'].map((k)=> [k, {
+            palette: {
+                // joyUI primary colors remain same for both dark and light mode
+                primary: {
+                    main: joyTheme.palette.primary[500],
+                    dark: joyTheme.palette.primary[600], //used by MUI in hover
```

Comment:
Interesting how MUI uses main/dark for main-bg and main-bg-hover.  As for color, I normally chooses the color name and let JoyUI do the mapping.  I am suspecting that the mappings may change when certain features are activated.  So, instead of fixed color, `500`/`600`, maybe use `palette.primary.solidBg` and `palette.primary.solidHoverBg`, instead?
![image](https://github.com/Caltech-IPAC/firefly/assets/11449642/8c2191ad-b118-4824-8330-4794e7ea21e2)

Reply:
I agreee

Reply:
I initially tried that but MUI theme doesn't allow CSS var strings:
```
Uncaught Error: MUI: Unsupported `var(--joy-palette-primary-500, #0B6BCB)` color.
The following formats are supported: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla(), color()
```

Reply:
No, I am suggesting to use this instead.
```
main: joyTheme.palette.primary.solidBg
dark: joyTheme.palette.primary.solidHoverBg
```

Reply:
Yes this is exactly what I tried. 'var(--joy-palette-primary-500, #0B6BCB)' is stored in joyTheme.palette.primary.solidBg

Reply:
From debugger, `joyTheme.palette.primary` is this:

```
{
    "50": "#EDF5FD",
    "100": "#E3EFFB",
    "200": "#C7DFF7",
    "300": "#97C3F0",
    "400": "#4393E4",
    "500": "#0B6BCB",
    "600": "#185EA5",
    "700": "#12467B",
    "800": "#0A2744",
    "900": "#051423",
    "plainColor": "var(--joy-palette-primary-500, #0B6BCB)",
    "plainHoverBg": "var(--joy-palette-primary-100, #E3EFFB)",
    "plainActiveBg": "var(--joy-palette-primary-200, #C7DFF7)",
    "plainDisabledColor": "var(--joy-palette-neutral-400, #9FA6AD)",
    "outlinedColor": "var(--joy-palette-primary-500, #0B6BCB)",
    "outlinedBorder": "var(--joy-palette-primary-300, #97C3F0)",
    "outlinedHoverBg": "var(--joy-palette-primary-100, #E3EFFB)",
    "outlinedActiveBg": "var(--joy-palette-primary-200, #C7DFF7)",
    "outlinedDisabledColor": "var(--joy-palette-neutral-400, #9FA6AD)",
    "outlinedDisabledBorder": "var(--joy-palette-neutral-200, #DDE7EE)",
    "softColor": "var(--joy-palette-primary-700, #12467B)",
    "softBg": "var(--joy-palette-primary-100, #E3EFFB)",
    "softHoverBg": "var(--joy-palette-primary-200, #C7DFF7)",
    "softActiveColor": "var(--joy-palette-primary-800, #0A2744)",
    "softActiveBg": "var(--joy-palette-primary-300, #97C3F0)",
    "softDisabledColor": "var(--joy-palette-neutral-400, #9FA6AD)",
    "softDisabledBg": "var(--joy-palette-neutral-50, #FBFCFE)",
    "solidColor": "var(--joy-palette-common-white, #FFF)",
    "solidBg": "var(--joy-palette-primary-500, #0B6BCB)",
    "solidHoverBg": "var(--joy-palette-primary-600, #185EA5)",
    "solidActiveBg": "var(--joy-palette-primary-700, #12467B)",
    "solidDisabledColor": "var(--joy-palette-neutral-400, #9FA6AD)",
    "solidDisabledBg": "var(--joy-palette-neutral-100, #F0F4F8)",
    "mainChannel": "11 107 203",
    "lightChannel": "199 223 247",
    "darkChannel": "18 70 123"
}

Reply:
Ok.  I get it now.  MUI validation does not allow `var(...)`.

Reply:
Yes. I can extract hash-color substring from solidBg and solidHoverBg if you want? But it is effectively the same thing

Reply:
How about this: this looks hacky but will do the job:
```js
function varColorStrToHex(theme, colorStr, defaultHex) {
    const regex = /^var(--joy-palette-(.+?)-(\d+), #([0-9a-fA-F]{6}))$/;
    const match = colorStr.match(regex);

    if (match) {
        const [, paletteName, shade, hexCode] = match;
        return theme?.palette?.[paletteName]?.[shade] ?? hexCode;
    } else {
       return defaultHex;
    }
}
```

Reply:
It's a lot of work to do something that may never be used.  But, since you did it already(it looks good), I think it's a good idea to use it.  Please add a comment to remind use why we're doing this.

Reply:
Ohk. BTW I just asked Bard to do it for me 😄  - it's fast this way to implement regex magic!